packages/headless/src/features/result-preview/result-preview-slice.ts

```diff
@@ -30,18 +34,13 @@ export const resultPreviewReducer = createReducer(
         state.isLoading = false;
       })
       .addCase(executeSearch.fulfilled, (state, action) => {
-        const {content, isLoading, uniqueId, contentURL} =
-          getResultPreviewInitialState();
-        state.content = content;
-        state.isLoading = isLoading;
-        state.uniqueId = uniqueId;
-        state.contentURL = contentURL;
-
+        resetPreviewContentState(state);
         state.resultsWithPreview = getUniqueIdsOfResultsWithHTMLVersion(
           action.payload.response.results
         );
       })
       .addCase(fetchMoreResults.fulfilled, (state, action) => {
+        resetPreviewContentState(state);
```

Comment:
I think we would also need to reset preview content state on `fetchPage.fullfilled`. 

I think you would be able to reproduce the same kind of problem reported on discuss with the results per page component, for example.

Reply:
Mh good catch! I thought about the `atomic-pager`, but I didn't do anything because I didn't see any reducer in place, so I assumed it was 'fine.

I now tested and confirm that the incorrect behavior is indeed present with the `atomic-pager`